python/lsst/summit/utils/efdUtils.py

```diff
@@ -661,3 +662,76 @@ def getTopics(client, toFind, caseSensitive=False):
             matches.append(topic)
 
     return matches
+
+
+def getCommands(client, commands, begin, end, prePadding, postPadding, timeFormat='python'):
+    """Retrieve the commands issued within a specified time range.
+
+    Parameters
+    ----------
+    client : `EfdClient`
+        The client object used to retrieve EFD data.
+    commands : `list`
+        A list of commands to retrieve.
+    begin : `astropy.time.Time`
+        The start time of the time range.
+    end : `astropy.time.Time`
+        The end time of the time range.
+    prePadding : `float`
+        The amount of time to pad before the begin time.
+    postPadding : `float`
+        The amount of time to pad after the end time.
+    timeFormat : `str`
+        One of 'pandas' or 'astropy' or 'python'. If 'pandas', the dictionary
+        keys will be pandas timestamps, if 'astropy' they will be astropy times
+        and if 'python' they will be python datetimes.
+
+    Returns
+    -------
+    commandTimes : `dict` [`time`, `str`]
+        A dictionary of the times at which the commands where issued. The type
+        that `time` takes is determined by the format key, and defaults to
+        python datetime.
+
+    Raises
+    ------
+    ValueError
+        Raise if there is already a command at a timestamp in the dictionary,
+        i.e. there is a collision.
+    """
+    if timeFormat not in ['pandas', 'astropy', 'python']:
+        raise ValueError(f"format must be one of 'pandas', 'astropy' or 'python', not {timeFormat=}")
+
+    commands = list(ensure_iterable(commands))
```

Comment:
Why "listify" here?  Can't the iterable be directly used in the `for` loop below?

Reply:
I hate that too, but `ensure_iterable` will just return an exhaustible generator if it got one of those, and I hate accidentally consuming things when looping over them to find they're now gone, so just tend to do that for safety when I know the list is small, mainly just out of habit. It's not necessary here, but it's also probably less harmful than not doing it, on balance.